[Rules] Adding actions to the rules details action menu (Part 2)

baileycash-elastic opened the pull request:
Continuing implementation of #199421

## Summary

This PR:
- moves snooze to a separate button similar to the mockup attached to the issue
- adds "run rule" and "update API key"

<img width="333" alt="Screenshot 2025-05-01 at 3 13 11 PM" src="https://github.com/user-attachments/assets/0e2a9a96-4940-45c2-a708-c0d3709080d9" />


<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

baileycash-elastic committed:
snooze and enable

baileycash-elastic committed:
close popover on snooze change

baileycash-elastic committed:
Merge branch 'main' into rules-199421

baileycash-elastic committed:
address errors

baileycash-elastic committed:
add other ops, change to context menu

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

baileycash-elastic committed:
merge in main

prodsecmachine commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/kibana/pr-checks/042c28b8-58af-43a1-a662-ede8ed9d2680?refs=d6c05090ab46684121dfc6e36895277a3309aa9e&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/kibana/pr-checks/042c28b8-58af-43a1-a662-ede8ed9d2680/license?refs=d6c05090ab46684121dfc6e36895277a3309aa9e&source=prChecksComment)

baileycash-elastic committed:
move snooze according to mockup

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

baileycash-elastic committed:
Merge branch 'main' into rules-199421-1

kdelemme commented on x-pack/solutions/observability/plugins/observability/public/hooks/use_update_api_key.ts:
```diff
@@ -0,0 +1,56 @@
+  const queryClient = useQueryClient();
+
+  const updateAPIKey = useMutation<string, string, { id: string }>(
+    ['runRule'],
```
duplicated key

kdelemme commented on x-pack/solutions/observability/plugins/observability/public/pages/rule_details/components/header_actions.tsx:
```diff
@@ -48,8 +52,18 @@ export function HeaderActions({
+    }
+  `;
+
   const { mutateAsync: enableRule } = useEnableRule();
   const { mutateAsync: disableRule } = useDisableRule();
+  const { mutateAsync: runRule } = useRunRule();
+  const { mutateAsync: updateAPIKey } = useUpdateAPIKey();
```
I don't think we use react query correctly here: https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync

since we don't handle the promises ourselves, we should rely on mutate.

kdelemme reviewed (commented):
Some comments about react hook query usage that i think we should fix

baileycash-elastic committed:
update duplicate key

baileycash-elastic committed:
use mutate over mutateAsync

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/298063)
* Commit: f2a05909981b137fd913c6090f3d22c9915f51cc

### Failed CI Steps
* [FTR Configs #51](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968d8d-2f3e-4e6c-9fd6-a97309795409)
* [FTR Configs #67](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968d8d-2f4a-4b9b-86c8-3ad0b2cae533)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968d8d-2f56-4fe4-b9df-a0cd5d68af1a)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968db5-9cb9-4bfb-b1a5-1067ff9572bc)
* [FTR Configs #104](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968d8d-2f66-4894-967f-40753c7f4e7b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968d8d-2f56-4fe4-b9df-a0cd5d68af1a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298063/jobs/01968d8d-2f56-4fe4-b9df-a0cd5d68af1a/artifacts/01968db5-53cd-424b-8dfa-c725d703448d) FTR Configs #<span></span>84 / ObservabilityApp Observability Rule Details page User permissions "after all" hook for "should not show the actions button if user has no permissions"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968db5-9cb9-4bfb-b1a5-1067ff9572bc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298063/jobs/01968db5-9cb9-4bfb-b1a5-1067ff9572bc/artifacts/01968dc4-1b63-4763-a5c3-73288cebcea5) FTR Configs #<span></span>84 / ObservabilityApp Observability Rule Details page User permissions "after all" hook for "should not show the actions button if user has no permissions"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968d8d-2f56-4fe4-b9df-a0cd5d68af1a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298063/jobs/01968d8d-2f56-4fe4-b9df-a0cd5d68af1a/artifacts/01968db5-53cd-4f6a-8627-fda588bd3f75) FTR Configs #<span></span>84 / ObservabilityApp Observability Rule Details page User permissions should show the rule edit and delete button if user has permissions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298063#01968db5-9cb9-4bfb-b1a5-1067ff9572bc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298063/jobs/01968db5-9cb9-4bfb-b1a5-1067ff9572bc/artifacts/01968dc4-1b63-4df8-a109-975fceeaeb26) FTR Configs #<span></span>84 / ObservabilityApp Observability Rule Details page User permissions should show the rule edit and delete button if user has permissions

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298063","number":298063,"commit":"f2a05909981b137fd913c6090f3d22c9915f51cc"}],"number":298063}
buildkite-pr-comment-->

baileycash-elastic committed:
Merge branch 'main' into rules-199421-1

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/298112)
* Commit: dbefa526a663c52a10b955866d57400902f6f437

### Failed CI Steps
* [FTR Configs #60](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968efa-9a86-4d06-8c62-985b534f6a8d)
* [FTR Configs #60](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968f22-ec03-4178-b227-0a7c51621cfc)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968efa-9a9b-4fc8-81f2-996b12db31ff)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968efa-9a86-4d06-8c62-985b534f6a8d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298112/jobs/01968efa-9a86-4d06-8c62-985b534f6a8d/artifacts/01968f22-d3cc-4b19-9463-5a3b8cf2f035) FTR Configs #<span></span>60 / ObservabilityApp Observability Rule Details page User permissions "after all" hook for "should not show the actions button if user has no permissions"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968f22-ec03-4178-b227-0a7c51621cfc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298112/jobs/01968f22-ec03-4178-b227-0a7c51621cfc/artifacts/01968f30-e114-4baa-ab9f-c5e07844e016) FTR Configs #<span></span>60 / ObservabilityApp Observability Rule Details page User permissions "after all" hook for "should not show the actions button if user has no permissions"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968efa-9a86-4d06-8c62-985b534f6a8d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298112/jobs/01968efa-9a86-4d06-8c62-985b534f6a8d/artifacts/01968f22-d3cd-49e1-857e-fca126bb23c7) FTR Configs #<span></span>60 / ObservabilityApp Observability Rule Details page User permissions should show the rule edit and delete button if user has permissions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298112#01968f22-ec03-4178-b227-0a7c51621cfc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298112/jobs/01968f22-ec03-4178-b227-0a7c51621cfc/artifacts/01968f30-e115-46b7-9d91-f9af22a880ef) FTR Configs #<span></span>60 / ObservabilityApp Observability Rule Details page User permissions should show the rule edit and delete button if user has permissions

### History
* :broken_heart: [Build #298063](https://buildkite.com/elastic/kibana-pull-request/builds/298063) failed f2a05909981b137fd913c6090f3d22c9915f51cc

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298112","number":298112,"commit":"dbefa526a663c52a10b955866d57400902f6f437"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298063","number":298063,"commit":"f2a05909981b137fd913c6090f3d22c9915f51cc"}],"number":298112}
buildkite-pr-comment-->

baileycash-elastic committed:
update test subject labels

baileycash-elastic committed:
Merge branch 'main' into rules-199421-1

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/298682)
* Commit: 27a25ae230dbc69ec9adb686004d474ec75a4211

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/d4e4e4ee4246b32f47cb933551a036b0257e75ca) | [after](https://github.com/elastic/kibana/commit/27a25ae230dbc69ec9adb686004d474ec75a4211) | diff |
| --- | --- | --- | --- |
| `observability` | 1280 | 1282 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d4e4e4ee4246b32f47cb933551a036b0257e75ca) | [after](https://github.com/elastic/kibana/commit/27a25ae230dbc69ec9adb686004d474ec75a4211) | diff |
| --- | --- | --- | --- |
| `observability` | 1.3MB | 1.3MB | +1.6KB |



### History
* :broken_heart: [Build #298112](https://buildkite.com/elastic/kibana-pull-request/builds/298112) failed dbefa526a663c52a10b955866d57400902f6f437
* :broken_heart: [Build #298063](https://buildkite.com/elastic/kibana-pull-request/builds/298063) failed f2a05909981b137fd913c6090f3d22c9915f51cc

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298682","number":298682,"commit":"27a25ae230dbc69ec9adb686004d474ec75a4211"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298112","number":298112,"commit":"dbefa526a663c52a10b955866d57400902f6f437"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298063","number":298063,"commit":"f2a05909981b137fd913c6090f3d22c9915f51cc"}],"number":298682}
buildkite-pr-comment-->

baileycash-elastic committed:
Merge branch 'main' into rules-199421-1

kdelemme commented:
One question, why the bell icons are in different state by default?

![Screenshot 2025-05-07 at 9 59 44 AM](https://github.com/user-attachments/assets/98e3fab2-3274-4858-85ef-0d8e234359ac)

baileycash-elastic commented on x-pack/solutions/observability/plugins/observability/public/pages/rule_details/components/header_actions.tsx:
```diff
@@ -118,76 +207,26 @@ export function HeaderActions({
+          <EuiButtonIcon
+            className="snoozeButton"
+            data-test-subj="snoozeRuleButton"
+            iconType={!getRuleHelpers(rule).isRuleSnoozed ? 'bellSlash' : 'bell'}
```
```suggestion
            iconType={getRuleHelpers(rule).isRuleSnoozed ? 'bellSlash' : 'bell'}
```

baileycash-elastic committed:
Update x-pack/solutions/observability/plugins/observability/public/pages/rule_details/components/header_actions.tsx

Reversing the condition to make the icon consistent with the interactive area of the status panel - and because the snooze state can be updated without updating the actual mute state

kdelemme reviewed (approved):
LGTM and tested locally.

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/14897800049

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/220458">](https://github.com/elastic/kibana/pull/220458)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @baileycash-elastic

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @baileycash-elastic

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @baileycash-elastic
